src/io/asyncsocket.c

```diff
@@ -457,12 +464,110 @@ static MVMint64 socket_handle(MVMThreadContext *tc, MVMOSHandle *h) {
     return (MVMint64)fd;
 }
 
+static MVMint64 get_sock_opt(MVMThreadContext *tc, MVMOSHandle *h, MVMint32 option) {
+    MVMIOAsyncSocketData *data   = h->body.data;
+    uv_handle_t          *handle = (uv_handle_t *)data->handle;
+    uv_os_fd_t            fd;
+    int                   s;
+    socklen_t             len;
+    int                   e;
+    MVMint64              output;
+
+    MVMROOT2(tc, handle, fd, {
```

Comment:
Is this `MVMROOT` neccesary?

Reply:
Nah, MVMROOT is only for garbage collectible objects (MVMCollectible)

Reply:
Ah, I'll keep that in mind